### PR TITLE
EVM Decoder

### PR DESCRIPTION
**Pull request by alex-miller-0**
**Blocked by #311**

* Adds EVM decoder option to general signing.
* Adds temporary bridge to migrate legacy ETH signing requests to use general signing so long as firmware is up to date.
* We should aim to deprecate the legacy ETH signing pathway (and remove `ethereum.ts`) after we go through a few more firmware updates. I will be opening similar PRs for BTC and ETH_MSG requests in the future.

### CODE REVIEW

File: src/ethereum.ts [hunk 1]
```diff
@@ -919,6 +920,58 @@ function get_rlp_encoded_preimage(rawTx, txType) {
   }
 }
 
+// ======
+// TEMPORARY BRIDGE
+// We are migrating from all legacy signing paths to a single generic
+// signing route. If users are attempting a legacy transaction request
+// against a Lattice on firmware v0.15.0 and above, we need to conver
```
Comment (douglance) [inline src/ethereum.ts:927]:
```suggestion
// against a Lattice on firmware v0.15.0 and above, we need to convert
```

**Review by douglance — APPROVED** [review]
I marked a couple really minor things.

File: src/ethereum.ts (hunk 1 above)
Comment (alex-miller-0) [reply src/ethereum.ts]: Force pushed in second commit: [0eeddd9](https://github.com/GridPlus/gridplus-sdk/pull/312/commits/0eeddd9c8b35092a0487e74ea7c24fed6bdb45de)

File: src/client.ts [hunk 2]
```diff
@@ -384,7 +385,7 @@ export class Client {
   public sign (opts: { data, currency: string }, _cb?: (err?: string, data?: SignData) => void, cachedData = null, nextCode = null): Promise<{ err?: string, data?: SignData }> {
     return new Promise((resolve, reject) => {
       const cb = promisifyCb(resolve, reject, _cb)
-      const { currency } = opts;
+      let { currency } = opts;
```
[the remark this answers is not in the document]
Comment (alex-miller-0) [reply src/client.ts]: Force pushed in second commit: [0eeddd9](https://github.com/GridPlus/gridplus-sdk/pull/312/commits/0eeddd9c8b35092a0487e74ea7c24fed6bdb45de)

**Pull request merged**